### DISCUSSION TITLE
Remove Theia Plugins from documentation

### DISCUSSION THREAD

**Discussion by JonasHelming** [eclipse-theia/theia#11579] [Improvements] [answered]
The documentation on how to develop Theia Plugins is outdated (https://github.com/eclipse-theia/theia-website/issues/313).

AFAIK, most people are using Theia extensions or VS Code extensions and Theia Plugins are a niche. Is anybody willing to fix the Theia Plugin Documentation (and the generator) short term? If not, I would propose to remove it from the documentation for now to not confuse new users. Please respond with your thoughts immediately. If there is no veto, I will take action on this on August 18th.

**Comment by vince-fugnitto** [comment]
@JonasHelming I'd be fine with removing the documentation for "theia plugins" (likely commenting it out) for the moment, as it is not as well supported as it's vscode's counterpart. I don't remember the latest state but they might not even work properly on Electron. For newcomers and in general we likely want to suggest the use of vscode extensions instead for the simplicity, and benefits over the theia plugins.

**Comment by zhangbaojia** [comment]
I think it's to remove some of the development related documentation and keep Theia and VSCode extensions different.I prefer to use Theia extension development, and my learning style is reverse development by analyzing Theia source code,Because the  "Using Thiea" documentation give me a hard time.

At least "OVERVIEW" it's good!

**Reply by JonasHelming** [reply]
Hi,

this is about "Theia Plugins" not "Theia extensions". Theia extensions are used by many adopters and the documentation will not be removed.

**Comment by msujew** [comment]
@JonasHelming @vince-fugnitto I agree. We don't have metrics on that, but I believe that barely anyone actually builds Theia plugins. Building a vscode extension is just plainly easier. Removing the documentation (and maybe even down the line the actual application support) sounds good to me.

**Comment by JonasHelming** [comment, marked as answer]
See here for hiding the plugin docu: https://github.com/eclipse-theia/theia-website/issues/314
and here for adding something about VS Code extensions: https://github.com/eclipse-theia/theia-website/issues/316

**Comment by thegecko** [comment]
Happy to remove these docs as I would recommend using the extension approach for any new project.

Removing the plugin support is a different concern, however and I'd want that to be maintained. Wherever we use plugins, we are happy to work with them without the docs (which I doubt we referenced anyway to be fair).
